### PR TITLE
famfs: Add fio stress test

### PR DESCRIPTION
**Pull request by arramesh42**
Run fio with multiple files filling up the entire filesystem

### CODE REVIEW

File: scripts/run_stress_tests.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
```
Comment (jagalactic) [inline scripts/run_stress_tests.sh:1]: I think this script should be hoisted up to the root of the repo, alongside run_smoke.sh and smoke_loop.sh - until we decide to move them all somewhere.

Should run with cwd=the root of the repo without passing in bin and scripts args. (I did this, with some small mixups, but then ran into the size calculation problem noted below).

File: scripts/stress_fio.sh [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+echo "SCRIPTS:  $SCRIPTS"
+
+
+NPROC=$(cat /proc/cpuinfo | awk '/^processor/{print $3}' | wc -l)
+NJOBS=$((NPROC - 1))
+
+# Get the device node name, ex: pmem0 from /dev/pmem0
+DEVICE=$(echo $DEV | awk -F '/' '{print $3}')
+# Get device size in bytes from ndctl cmd
+DEV_SIZE_BYTES=$(sudo ndctl list | grep -B2 $DEVICE | grep \"size\" | awk -F ':' '{print $2}' | awk -F ',' '{print $1}')
```
[the remark this answers is not in the document]
Comment (jagalactic) [reply scripts/stress_fio.sh:59]: Also, you should get this from syses - which is simpler and less brittle than the method above. famfs_lib.c does this...

**Comment by jagalactic** [comment]
Please also explain exactly how to run this when you update the PR, to shorten the test/review cycle.

File: scripts/run_stress_tests.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+# Allow these variables to be set from the environment
+if [ -z "$MPT" ]; then
+    MPT=/mnt/famfs
+fi
+if [ -z "$DEV" ]; then
+    DEV="/dev/pmem0"
+fi
+
+# Verify that this script is running as root
```
Comment (jagalactic) [inline scripts/run_stress_tests.sh:19]: This script doesn't need to be run as root, but it does need sudo (which is the next test). So drop this test.

File: scripts/stress_fio.sh [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+#echo "last_proc=$last_proc"
+
+
+for (( cpu = 0; cpu < $NJOBS; cpu++ ))
+do
+    for (( fnum = 0; fnum < $FILES_PER_THRD; fnum++ ))
+    do
+        fname="$NAME.$cpu.$fnum"
+        echo "Creating famfs file $MPT/$fname"
+        sudo $BIN/famfs creat -v -s $FILE_SIZE $MPT/$fname
```
Comment (jagalactic) [inline scripts/stress_fio.sh:96]: Should fail if these "creats" fail, which they do if it'd /dev/dax.

File: scripts/stress_fio.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jagalactic) [reply scripts/stress_fio.sh:59]: Actually scripts/test_funcs.sh has a bash function get_device_size() that will handle either pmem or dax via sysfs

[the remark this answers is not in the document]
Comment (arramesh42) [reply scripts/stress_fio.sh]: I had tested with dax devices also and it had worked for me. 
I have now made dax device as the default device type and also getting the device size from sysfs. this should take care of zero size files. Added explicit check to verify file size is not zero.

File: scripts/run_stress_tests.sh (hunk 1 above)
Comment (arramesh42) [reply scripts/run_stress_tests.sh]: Ok, moved it to the "root of the repo" and made necessary changes to paths.

File: scripts/run_stress_tests.sh (hunk 3 above)
Comment (arramesh42) [reply scripts/run_stress_tests.sh]: Ok removed.

File: scripts/stress_fio.sh (hunk 4 above)
Comment (arramesh42) [reply scripts/stress_fio.sh]: Ok. Updated.

**Comment by arramesh42** [comment]
This PR adds a fio based stress test. It expects the fio binary to be installed in the system. 
This test can be run using the command:

"**_make stress_tests_**"  => this runs the test for a default runtime of 60 seconds on /dev/dax0.0

It can also be run using the below command, where we can specify custom fio binary path (using -f option) and custom runtime using -r and custom device /dev/dax1.0.

"_**sudo ./run_stress_tests.sh -d /dev/dax1.0 -f /usr/local/Repos/Fio/fio/fio -r 300**_ "                       <== this runs the fio test on /dev/dax1.0 using the fio available at the location /usr/Repo/... for a runtime of 5 mins. 
Note: -f expects the full binary path and not the directory. Also as mentioned above, "-f " is optional, this tests uses system installed fio if it is available in the system.

**Pull request merged**